storage/src/vespa/storage/storageserver/statemanager.cpp

```diff
@@ -392,36 +390,28 @@ StateManager::onGetNodeState(const api::GetNodeStateCommand::SP& cmd)
 {
     bool sentReply = false;
     if (cmd->getSourceIndex() != 0xffff) {
-        sentReply = sendGetNodeStateReplies(framework::MilliSecTime(0),
-                                            cmd->getSourceIndex());
+        sentReply = sendGetNodeStateReplies(cmd->getSourceIndex());
     }
     std::shared_ptr<api::GetNodeStateReply> reply;
     {
         std::unique_lock guard(_stateLock);
         const bool is_up_to_date = (_controllers_observed_explicit_node_state.find(cmd->getSourceIndex())
                                     != _controllers_observed_explicit_node_state.end());
-        if (cmd->getExpectedState() != nullptr
+        if ((cmd->getExpectedState() != nullptr)
             && (*cmd->getExpectedState() == *_nodeState || sentReply)
             && is_up_to_date)
         {
-            int64_t msTimeout = vespalib::count_ms(cmd->getTimeout());
-            LOG(debug, "Received get node state request with timeout of "
-                       "%" PRId64 " milliseconds. Scheduling to be answered in "
-                       "%" PRId64 " milliseconds unless a node state change "
-                       "happens before that time.",
-                msTimeout, msTimeout * 800 / 1000);
-            TimeStateCmdPair pair(
-                    _component.getClock().getTimeInMillis()
-                    + framework::MilliSecTime(msTimeout * 800 / 1000),
-                    cmd);
-            _queuedStateRequests.emplace_back(std::move(pair));
+            vespalib::duration timeout = cmd->getTimeout();
+            if (timeout == vespalib::duration::max()) timeout = 24h; //balder: Dirty temporary hack
```

Comment:
Realistically this could be capped to a much smaller value—such as 5 minutes—and thus be promoted from being a hack. With default options, the cluster controller operates with a timeout of 120 seconds for node state long-polls.

Reply:
Fixed